conan/tools/sbom/cyclonedx.py

```diff
@@ -28,25 +29,32 @@ def cyclonedx_1_4(conanfile, name=None, add_build=False, add_tests=False, **kwar
     from datetime import datetime, timezone
     graph = conanfile.subgraph
 
-    has_special_root_node = not (getattr(graph.root.ref, "name", False) and getattr(graph.root.ref, "version", False) and getattr(graph.root.ref, "revision", False))
+    has_special_root_node = not (getattr(graph.root.ref, "name", False)
+                                 and getattr(graph.root.ref, "version", False)
+                                 and getattr(graph.root.ref, "revision", False))
     special_id = str(uuid.uuid4())
 
     name_default = getattr(graph.root.ref, "name", False) or "conan-sbom"
-    name_default += f"/{graph.root.ref.version}" if bool(getattr(graph.root.ref, "version", False)) else ""
-    nodes = [node for node in graph.nodes if (node.context == "host" or add_build) and (not node.test or add_tests)]
+    name_default += f"/{graph.root.ref.version}" if getattr(graph.root.ref, "version", False) else ""
+
+    def should_add_node(n):
```

Comment:
This is the fix, making sure no node is added to ``dependsOn`` that is not checked against this condition.